keycloak.d.ts

```diff
@@ -0,0 +1,348 @@
+import * as express from 'express'
+
+/**
+ * The JavaScript module is exported as a single function, but for TypeScript we
+ * need to export the function and a set of interfaces so developers can assign
+ * types such as Grant, Token, etc. to variables in their own code.
+ * 
+ * To achieve this we export "KeycloakConnect" that references a namespace
+ * containing our typings, and a static instance exposing the constructor
+ */
+declare const KeycloakConnect: KeycloakConnectStatic
+export = KeycloakConnect
+
+interface KeycloakConnectStatic {
+  new (options: KeycloakConnect.KeycloakOptions, config: KeycloakConnect.KeycloakConfig): KeycloakConnect.Keycloak
+}
+
+declare namespace KeycloakConnect {
+
+  interface KeycloakConfig {
+    'confidential-port': string|number
+    'auth-server-url': string
+    'resource': string
+    'ssl-required': string
+    'bearer-only'?: boolean
+    realm: string
+  }
+
+  interface KeycloakOptions {
+    scope?: string
+    store?: any
+    cookies?: boolean
+  }
+
+  interface GrantProperties {
+    access_token?: string
+    refresh_token?: string
+    id_token?: string
+    expires_in?: string
+    token_type?: string
+  }
+
+  interface Token {
```

Comment:
Technically these _should_ be classes, but it causes lines 11 and 12 not work, so using interfaces is a decent compromise.

Reply:
For my own edification, why do you say they should be classes?